### PR TITLE
Update Qos SDK with new QosResult format

### PR DESCRIPTION
**Pull request by yzhao174**
1. Update Qos SDK with new QosResult format
2. Return sorted list based on Ping result
3. Sent timeout 
4. Update event namespace
5. Use WriteTelemetryEvents instead of WriteEvents

### CODE REVIEW

File: build/Windows/XPlatCppWindows.vcxproj [hunk 1]
```diff
@@ -168,7 +167,7 @@
     <ClInclude Include="$(SdkSourceDir)\include\playfab\PlayFabSpinLock.h" />
     <ClInclude Include="$(SdkSourceDir)\include\playfab\PlayFabTransportHeaders.h" />
     <ClInclude Include="$(SdkSourceDir)\include\playfab\PlayFabJsonHeaders.h" />
-    <ClInclude Include="$(SdkSourceDir)\include\playfab\QoS\DataCenterResult.h" />
+    <ClInclude Include="..\..\code\include\playfab\QoS\RegionResult.h" />
```
Comment (toddbellMSFT) [inline build/Windows/XPlatCppWindows.vcxproj]: making these relative paths seems weird, do we have precedence for this in the file already?

Comment (yzhao174) [reply build/Windows/XPlatCppWindows.vcxproj]: Good catch. This is a rename from DataCenterResult to RegionResult. It doesn't change the path in a consistent way. I will fix it

File: code/include/playfab/QoS/RegionResult.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            RegionResult() = delete;
+
+            // The datacenter region
+            PlayFab::MultiplayerModels::AzureRegion region;
+
+            // Average latency to reach the data center
+            int latencyMs;
+
+            // Last error code recieved while pinging
+            int errorCode;
```
Comment (toddbellMSFT) [inline code/include/playfab/QoS/RegionResult.h]: if all the members are public, should this just be a struct? Or should these members be private with public getters/setters?

File: code/source/playfab/QoS/PlayFabQoSApi.cpp [hunk 3]
```diff
@@ -22,6 +23,70 @@ namespace PlayFab
...
+        }
+
+        void OnWriteEventsResult(int httpCode, std::string result, std::unique_ptr<CallRequestContainerBase> reqContainer)
+        {
+            CallRequestContainer& container = static_cast<CallRequestContainer&>(*reqContainer);
+
+            WriteEventsResponse outResult;
+            if (ValidateResult(outResult, container))
+            {
+
```
Comment (toddbellMSFT) [inline code/source/playfab/QoS/PlayFabQoSApi.cpp]: nitpick: excess whitespace

File: test/TestApp/TestApp.cpp [hunk 4]
```diff
@@ -44,16 +44,16 @@ namespace PlayFabUnit
         // Add PlayFab API tests.
         PlayFabApiTest pfApiTest;
         pfApiTest.SetTitleInfo(testInputs);
-        testRunner.Add(pfApiTest);
+        //testRunner.Add(pfApiTest);
```
Comment (pgilmorepf) [inline test/TestApp/TestApp.cpp]: I do not approve of disabling the tests for this feature.
These tests must execute, test the QoS feature, and pass in order for this checkin to be accepted.

Comment (yzhao174) [reply test/TestApp/TestApp.cpp]: Oops. This was disabled while debugging.

File: code/include/playfab/QoS/RegionResult.h (hunk 2 above)
Comment (yzhao174) [reply code/include/playfab/QoS/RegionResult.h]: Change to struct in this case

File: code/source/playfab/QoS/PlayFabQoSApi.cpp [hunk 5]
```diff
@@ -22,6 +23,69 @@ namespace PlayFab
 {
     namespace QoS
     {
+        bool ValidateResult(PlayFabResultCommon& resultCommon, CallRequestContainer& container)
```
Comment (MichaelMackendMS) [inline code/source/playfab/QoS/PlayFabQoSApi.cpp]: Can the container be const&?

Comment (yzhao174) [reply code/source/playfab/QoS/PlayFabQoSApi.cpp]: No, the method call below "FromJson" expect a non-const&

**Comment by yzhao174** [comment]
Change is already merged into SDKGenerator

**Pull request closed without merging**